test/Main.hs

```diff
@@ -0,0 +1,75 @@
+module Main where
+
+import Control.Monad (when)
+import Data.List (isSuffixOf)
+import System.Directory (doesFileExist, findExecutable, removeFile)
+import System.Exit (ExitCode (ExitSuccess), die)
+import System.Process (proc, readCreateProcessWithExitCode)
+import Test.Hspec (Expectation, Spec, afterAll_, describe, hspec, it, shouldBe, shouldSatisfy)
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = do
+  describe "hiedb" $ do
+    afterAll_ removeTestDb $ do
+      describe "index" $
+        it "indexes testing project .hie files" $ do
+          runHieDbCli ["index" , ".hie", "--quiet"]
+            `suceedsWithStdin` ""
+
+      describe "ls" $
+        it "lists the indexed modules" $ do
+          runHieDbCli ["ls"]
+            `succeedsWithStdinSatisfying` (\stdin -> do
+              let inlines = lines stdin
+              length inlines `shouldBe` 2
+              (inlines !! 0) `shouldSatisfy` isSuffixOf ".hie/Foo.hie\tFoo\thiedb-0.1.0.0-inplace-test-lib"
```

Comment:
Not suer how brittle these kinds of tests will be.
E.g. here I had to do this trick with custom predicates, because the indexed hie file paths are full paths which will differe from system to system.

Reply:
It will be probably best to switch to some form of golden testing, with expected stdout / stderr for each command run in separate file. I'll look into it next